conformance/lang/expressions/string-template-expr/string_template_expr.balt

```diff
@@ -0,0 +1,323 @@
+    io:println(temp1); // @output b1
+    string temp2 = string `v1 v2`;
+    io:println(temp2); // @output v1 v2
+    string temp3 = string `$ $ $ $ $ $`;
+    io:println(temp3); // @output $ $ $ $ $ $
+    string temp4 = string `$$$$`;
+    io:println(temp4); // @output $$$$
+    string temp5 = string `$v1$$v2$`;
+    io:println(temp5); // @output $v1$$v2$
+    string temp6 = string `{}`;
+    io:println(temp6); // @output {}
+}
+
+Test-Case: error
+Description: Test every character that is not part of an interpolation is interpreted as a literal character.
+Fail-Issue: ballerina-platform/ballerina-lang#35515
+Labels: string, function-defn, string-template-expr
+
+function errorFunction() {
+    string _ = string `${`;
+}
+
+function f1() {
+    string name = "Ballerina";
+    string _ = string `${name}`;
+}
+
+function f2() {
+    string _ = string `${`;
+}
+
+
+Test-Case: error
+Description: Test every character that is not part of an interpolation is interpreted as a literal character.
+Labels: string, function-defn, string-template-expr
+
+function errorFunction() {
+    string _ = string `${`}`; @error backtick as a expression
+    string _ = string ```; // @error backtick within BacktickString
+    string _ = string ````; // @error backtick within BacktickString
+    string _ = string ``````; // @error backtick within BacktickString
+}
+
+Test-Case: output
+Description: Test string-template-expr is evaluated by evaluating the expression in each interpolation in the order in
+which they occur.
+Labels: string, function-defn, function-call-expr, string-template-expr, int, boolean
+
+function init() {
+    string a1 = "a1";
+    string b1 = "b1";
+    string temp1 = string `${a1} ${b1}`;
+    io:println(temp1); // @output a1 b1
+    int a2 = 12;
+    string temp2 = string `${a2} ${b1+ a1}`;
+    io:println(temp2); // @output 12 b1a1
+    int a3 = 13;
+    int b2 = 15;
+    string temp3 = string `$ $ $ ${a2 + a3} $ $ $ ${b2 + a2}`;
+    io:println(temp3); // @output $ $ $ 25 $ $ $ 27
+    boolean a4 = true;
+    string temp4 = string `$ $ $ ${a2 + a3} $ $ $ ${a4}`;
+    io:println(temp4); // @output $ $ $ 25 $ $ $ true
+    string temp5 = string `$ $ $ ${f1()} $ $ $ ${f2()} $ $ $ ${f3()}`;
+    io:println(temp5); // @output $ $ $ 10 $ $ $ 10 $ $ $ 10
+    string temp6 = string `${"c"}${"s"}`;
+    io:println(temp6); // @output cs
+}
+
+function f1() returns int {
+    return 10;
+}
+
+function f2() returns int {
+    return 10;
+}
+
+function f3() returns int {
+    return 10;
+}
+
+Test-Case: output
+Description: Test literal ` included in string template.
+Labels: string, function-defn, string-template-expr
+
+function init() {
+    string temp1 = string `${"`"}`;
+    io:println(temp1); // @output `
+}
+
+Test-Case: error
+Description: Test interpolation without expression
+Labels: string, function-defn, string-template-expr
+
+function errorFunction() {
+    string _ = string `${}`; // @error missing expression
+}
```

Comment:
Please add a test or two to explicitly check "... and converting the result of the each evaluation to a string using the [ToString](https://ballerina.io/spec/lang/master/#ToString) abstract operation with the direct style".